src/Spout/Writer/XLSX/Helper/StyleHelper.php

```diff
@@ -84,13 +84,22 @@ protected function getFontsSectionContent()
      */
     protected function getFillsSectionContent()
     {
-        return <<<EOD
-<fills count="1">
-    <fill>
-        <patternFill patternType="none"/>
-    </fill>
-</fills>
-EOD;
+        $registeredStyles = $this->getRegisteredStyles();
+        $fillsCount = count($registeredStyles);
+        $content = sprintf('<fills count="%d">', $fillsCount);
+        foreach ($registeredStyles as $style) {
+            if ($style->shouldApplyBackgroundColor()) {
```

Comment:
should you add a check either here or in the function directly: && $style->getBackgroundColor() != DEFAULT_BG_COLOR?